spring-ai-alibaba-nl2sql/chat/src/main/java/com/alibaba/cloud/ai/service/Nl2SqlService.java

```diff
@@ -17,106 +17,24 @@
 
 import com.alibaba.cloud.ai.dbconnector.DbAccessor;
 import com.alibaba.cloud.ai.dbconnector.DbConfig;
-import com.alibaba.cloud.ai.dbconnector.MdTableGenerator;
-import com.alibaba.cloud.ai.dbconnector.bo.DbQueryParameter;
-import com.alibaba.cloud.ai.dbconnector.bo.ResultSetBO;
-import com.alibaba.cloud.ai.prompt.PromptHelper;
-import com.alibaba.cloud.ai.schema.SchemaDTO;
-import com.alibaba.cloud.ai.util.DateTimeUtil;
-import com.alibaba.cloud.ai.util.MarkdownParser;
-import com.google.gson.Gson;
-import com.google.gson.reflect.TypeToken;
-import org.springframework.ai.document.Document;
+import com.alibaba.cloud.ai.service.base.BaseNl2SqlService;
+import com.alibaba.cloud.ai.service.base.BaseSchemaService;
+import com.alibaba.cloud.ai.service.base.BaseVectorStoreService;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
 import org.springframework.stereotype.Service;
 
-import java.time.LocalDate;
-import java.util.*;
-import java.util.stream.Collectors;
-
-import static com.alibaba.cloud.ai.prompt.PromptHelper.buildMixSelectorPrompt;
-
+@ConditionalOnProperty(prefix = "spring.ai.vectorstore.analytic", name = "enabled", havingValue = "true",
+		matchIfMissing = true)
```

Comment:
Consider introduce separate autoconfigure and starter for bean and dependency management when there're more implementation choices.